Expose a function to return the LMDB version

I want to expose a very simple function that returns the major, minor, and patch versions of the underlying LMDB library.

```rust
fn heed::lmdb_version() -> (major, minor, patch);
```

Getting error while `cargo build`

error: failed to run custom build command for `lmdb-master-sys v0.1.0 (C:\Projects\Rust\Github\199-fix\lmdb-master-sys)`

Caused by:
  process didn't exit successfully: `C:\Projects\Rust\Github\199-fix\target\debug\build\lmdb-master-sys-4cded00d7d715647\build-script-build` (exit code: 1)
  --- stdout
  TARGET = Some("x86_64-pc-windows-msvc")
  OPT_LEVEL = Some("0")
  HOST = Some("x86_64-pc-windows-msvc")
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  DEBUG = Some("true")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-msvc
  CFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_msvc
  CFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-msvc
  CFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_msvc
  CFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-msvc
  CFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_msvc
  CFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-msvc
  CC_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_msvc
  CC_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-msvc
  CFLAGS_x86_64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_msvc
  CFLAGS_x86_64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  running: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-W4" "-FoC:\\Projects\\Rust\\Github\\199-fix\\target\\debug\\build\\lmdb-master-sys-543b8494d1ec48d1\\out\\c5dfcf082afb74ec-mdb.o" "-c" "C:\\Projects\\Rust\\Github\\199-fix\\lmdb-master-sys\\lmdb\\libraries\\liblmdb\\mdb.c"
  mdb.c
  c1: fatal error C1083: Cannot open source file: 'C:\Projects\Rust\Github\199-fix\lmdb-master-sys\lmdb\libraries\liblmdb\mdb.c': No such file or directory  
  exit code: 2

  --- stderr


  error occurred: Command "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-W4" "-FoC:\\Projects\\Rust\\Github\\199-fix\\target\\debug\\build\\lmdb-master-sys-543b8494d1ec48d1\\out\\c5dfcf082afb74ec-mdb.o" "-c" "C:\\Projects\\Rust\\Github\\199-fix\\lmdb-master-sys\\lmdb\\libraries\\liblmdb\\mdb.c" with args "cl.exe" did not execute successfully (status code exit code: 2).


warning: build failed, waiting for other jobs to finish...

--- a/heed/src/lib.rs
+++ b/heed/src/lib.rs
@@ -57,7 +57,8 @@ mod mdb;
 mod reserved_space;
 mod txn;
 
-use std::{error, fmt, io, result};
+use std::ffi::CStr;
+use std::{error, fmt, io, mem, result};
 
 use heed_traits as traits;
 pub use {bytemuck, byteorder, heed_types as types};
@@ -78,6 +79,49 @@ pub use self::mdb::flags::Flag;
 pub use self::reserved_space::ReservedSpace;
 pub use self::traits::{BoxedError, BytesDecode, BytesEncode};
 pub use self::txn::{RoTxn, RwTxn};
+
+/// The underlying LMDB library version information.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct LmdbVersion {
+    /// The library version as a string.
+    pub string: &'static str,
+    /// The library major version number.
+    pub major: i32,
+    /// The library minor version number.
+    pub minor: i32,
+    /// The library patch version number.
+    pub patch: i32,
+}
+
+/// Return the LMDB library version information.
+///
+/// ```
+/// use heed::{lmdb_version, LmdbVersion};
+///
+/// let expected = LmdbVersion {
+///     string: "LMDB 0.9.70: (December 19, 2015)",
+///     major: 0,
+///     minor: 9,
+///     patch: 70,
+/// };
+/// assert_eq!(lmdb_version(), expected);
+/// ```
+pub fn lmdb_version() -> LmdbVersion {
+    let mut major = mem::MaybeUninit::uninit();
+    let mut minor = mem::MaybeUninit::uninit();
+    let mut patch = mem::MaybeUninit::uninit();
+
+    unsafe {
+        let string_ptr =
+            mdb::ffi::mdb_version(major.as_mut_ptr(), minor.as_mut_ptr(), patch.as_mut_ptr());
+        LmdbVersion {
+            string: CStr::from_ptr(string_ptr).to_str().unwrap(),
+            major: major.assume_init(),
+            minor: minor.assume_init(),
+            patch: patch.assume_init(),
+        }
+    }
+}
 
 /// An error that encapsulates all possible errors in this crate.
 #[derive(Debug)]

--- a/heed/src/mdb/lmdb_ffi.rs
+++ b/heed/src/mdb/lmdb_ffi.rs
@@ -6,7 +6,7 @@ pub use ffi::{
     mdb_env_get_fd, mdb_env_get_flags, mdb_env_info, mdb_env_open, mdb_env_set_mapsize,
     mdb_env_set_maxdbs, mdb_env_set_maxreaders, mdb_env_stat, mdb_env_sync, mdb_filehandle_t,
     mdb_get, mdb_put, mdb_reader_check, mdb_stat, mdb_txn_abort, mdb_txn_begin, mdb_txn_commit,
-    MDB_cursor, MDB_dbi, MDB_env, MDB_envinfo, MDB_stat, MDB_txn, MDB_val, MDB_APPEND,
+    mdb_version, MDB_cursor, MDB_dbi, MDB_env, MDB_envinfo, MDB_stat, MDB_txn, MDB_val, MDB_APPEND,
     MDB_CP_COMPACT, MDB_CREATE, MDB_CURRENT, MDB_RDONLY, MDB_RESERVE,
 };
 use lmdb_master_sys as ffi;

